tools/generate-module-dependencies/internal/apply-replaces.go

```diff
@@ -67,8 +69,8 @@ func getMarkers(fileType types.FileType) (startMarker, endMarker string, err err
 
 // Upserts the generated block using the markers, or lack thereof, as a guide
 func upsertGeneratedBlock(targetContent, replacement, startMarker, endMarker string) (string, error) {
-	startIdx := strings.Index(targetContent, startMarker)
-	startFound := startIdx != -1
+	lineStart := strings.Index(targetContent, startMarker)
+	startFound := lineStart != -1
```

Comment:
I wonder if it could be simpler if we:
- Used [Split](https://pkg.go.dev/strings#Split) to split the targetContent into slice of lines
- Used `Contains` to find the line with start and end marker
- Replaced the middle part of the slice
- Used [Join](https://pkg.go.dev/strings#Join) to join it all back into text

This way we don't need to have the complexity of `lineStart` etc.